R/stroke.R

```diff
@@ -191,28 +193,42 @@ get_best_link <- function(angles, links, angle_threshold = 0) {
 }
 
 #' @noRd
-cross_check_links <- function(best_links, flow_mode = FALSE) {
+check_reciprocal <- function(links, best_links, colname) {
+    # keep only indices that are between 0 and the length of best_links
+    valid_indices <- best_links[, colname]
+    valid_indices <- valid_indices[valid_indices > 0 & valid_indices <= nrow(best_links)]
+    # find the best link of the best links 
+    bl <- best_links[valid_indices, , drop = FALSE]
```

Comment:
Why do we need to check for valid indices here? I think all best links should already fulfil this condition?

Reply:
While I was refactoring, `cross_check_links()` was stalling most likely because of `check_reciprocal()` so I was checking for valid indices. But that is not needed indeed.